packages/backend/src/open_payments/wallet_address/model.ts

```diff
@@ -229,3 +230,15 @@ export abstract class WalletAddressSubresource extends BaseModel {
   QueryBuilderType!: SubresourceQueryBuilder<this>
   static QueryBuilder = SubresourceQueryBuilder
 }
+
+export function throwIfMissingWalletAddress(
+  deps: { logger: Logger },
+  resource: WalletAddressSubresource
+): asserts resource is WalletAddressSubresource &
+  Required<Pick<WalletAddressSubresource, 'walletAddress'>> {
+  if (!resource.walletAddress) {
+    const errorMessage = `${resource.$modelClass.name} does not have wallet address. This should be investigated.`
+    deps.logger.error({ id: resource.id }, errorMessage)
+    throw new Error(errorMessage)
+  }
+}
```

Comment:
Function checks whether incoming payment, quote or outgoing payment has defined `walletAddress` property. Maybe the log message could be less about investigation though? The only way this would happen if we don't add `walletAddress` wasn't added to the `withGraphFetched()` function when getting the resource.

Ideally this would all be resolved with types (ie. we could specify that if `walletAddress` is in the `withGraphFetched` then we can have that directly in the type -> but that would be a pretty big change, although it would be handy I think?

Reply:
> Maybe the log message could be less about investigation though

I guess the pertinent thing is that its just (I assume) completely unexpected.

Is this based off any experience with a missing wallet address? I dont see this used so not exactly sure what the intention is and wondering why we need it.

Reply:
It's only to make typescript happy:

The  `WalletAddressSubresource` model:
```ts
export abstract class WalletAddressSubresource extends BaseModel {
  ...
  public readonly walletAddressId!: string
  public walletAddress?: WalletAddress
```

In order to have `walletAddress` defined we need to include it in any `withGraphFetched` requests. However, Objection of course wouldn't know anything about typescript types, so we're left with `walletAddress?: WalletAddress`. In our routes, we need the `walletAddress` to be defined to do e.g.:
```
  ctx.body = incomingPayment.toOpenPaymentsTypeWithMethods(
    incomingPayment.walletAddress,
    streamCredentials
  )
```

when calling function we just assert that the wallet subresource will for sure have walletAddress defined.

Reply:
Ah, I see. For some reason I didn't see the usage until now.

Would we save ourselves some trouble by doing it in the service instead? where we are specifying `.withGraphFetched('[asset, walletAddress]')`? Presumably it's always a problem if we dont find the `walletAddress` in such cases.

Or is it acting as a typegaurd that wouldn't work if used like that? It kinda looks that way but commenting out it's usage doesn't seem to result in any errors for me.